library/src/test/java/io/ona/kujaku/views/KujakuMapViewTest.java

```diff
@@ -0,0 +1,214 @@
+        assertNull(kujakuMapView.dropPoint(new LatLng(45d, 45d)));
+    }
+
+    @Test
+    public void addPointShouldMakeAddViewVisible() {
+        ImageButton addBtn = kujakuMapView.findViewById(R.id.imgBtn_mapview_locationAdditionBtn);
+        LinearLayout buttonsLayout = kujakuMapView.findViewById(R.id.ll_mapview_locationSelectionBtns);
+
+        assertEquals(View.GONE, addBtn.getVisibility());
+        kujakuMapView.addPoint(false, new AddPointCallback() {
+            @Override
+            public void onPointAdd(JSONObject jsonObject) {
+                // Do nothing
+            }
+
+            @Override
+            public void onCancel() {
+                // Do nothing
+            }
+        });
+
+        assertEquals(View.VISIBLE, addBtn.getVisibility());
+        assertEquals(View.GONE, buttonsLayout.getVisibility());
+    }
+
+    @Test
+    public void addPointShouldPerformActionsOnClick() {
+        ImageButton addBtn = kujakuMapView.findViewById(R.id.imgBtn_mapview_locationAdditionBtn);
+        Button doneAddingPoint = kujakuMapView.findViewById(R.id.btn_mapview_locationSelectionBtn);
+        LinearLayout buttonsLayout = kujakuMapView.findViewById(R.id.ll_mapview_locationSelectionBtns);
+
+        String isPointAddCalled = "isPointAddCalled";
+
+        HashMap<String, Object> states = new HashMap<>();
+        states.put(isPointAddCalled, false);
+
+        kujakuMapView.addPoint(false, new AddPointCallback() {
+            @Override
+            public void onPointAdd(JSONObject jsonObject) {
+                states.put(isPointAddCalled, true);
+            }
+
+            @Override
+            public void onCancel() {
+                // Do nothing
+            }
+        });
+
+        assertEquals(View.GONE, buttonsLayout.getVisibility());
+
+        addBtn.callOnClick();
+
+        assertEquals(View.VISIBLE, buttonsLayout.getVisibility());
+        assertEquals(View.GONE, addBtn.getVisibility());
+        assertTrue(kujakuMapView.isCanAddPoint());
+
+        doneAddingPoint.callOnClick();
+
+        assertTrue((boolean) states.get(isPointAddCalled));
+        assertEquals(View.GONE, buttonsLayout.getVisibility());
+        assertEquals(View.VISIBLE, addBtn.getVisibility());
+        assertFalse(kujakuMapView.isCanAddPoint());
+    }
+
+    @Test
+    public void showCurrentLocationBtnShouldChangeVisibleWhenCalled() {
+        ImageButton currentLocationBtn = kujakuMapView.findViewById(R.id.ib_mapview_focusOnMyLocationIcon);
+        assertEquals(View.GONE, currentLocationBtn.getVisibility());
+
+        kujakuMapView.showCurrentLocationBtn(true);
+        assertEquals(View.VISIBLE, currentLocationBtn.getVisibility());
+
+        kujakuMapView.showCurrentLocationBtn(false);
+        assertEquals(View.GONE, currentLocationBtn.getVisibility());
+    }
+
+    @Test
+    public void focusOnUserLocationShouldChangeTargetIconWhenGivenCalled() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
Does this really test that the target icon is changed?

Reply:
Added the assert for icon change